core/templates/pages/admin-page/misc-tab/admin-misc-tab.directive.ts

```diff
@@ -161,6 +167,119 @@ angular.module('oppia').directive('adminMiscTab', [
+            var emptyDiv = document.createElement('div');
+            var outputElement = angular.element(emptyDiv);
+            // We need to append the element with a script tag so that Mathjax
+            // can typeset and convert this element. The typesetting is not
+            // possible if we don't add a script tag. The code below is similar
+            // to how the math equations are rendered in the mathjaxBind
+            // directive (see mathjax-bind.directive.ts).
+            var $script = angular.element(
+              '<script type="math/tex">'
+            ).html(inputLatexString === undefined ? '' : inputLatexString);
+            outputElement.html('');
+            outputElement.append($script);
+            // Naturally MathJax works asynchronously, but we can add processes
+            // which we want to happen synchronously into the MathJax Hub Queue.
+            MathJax.Hub.Queue(['Typeset', MathJax.Hub, outputElement[0]]);
+            MathJax.Hub.Queue(function() {
+              var svgString = (
+                outputElement[0].getElementsByTagName('svg')[0].outerHTML);
+              var cleanedSvgString = (
+                ImageUploadHelperService.cleanMathExpressionSvgString(
+                  svgString));
+              var dimensions = (
+                ImageUploadHelperService.
+                  extractDimensionsFromMathExpressionSvgString(
+                    cleanedSvgString));
+              var dataURI = (
+                'data:image/svg+xml;base64,' + btoa(cleanedSvgString));
+              var invalidTagsAndAttributes = (
+                ImageUploadHelperService.getInvalidSvgTagsAndAttrs(dataURI));
+              var tags = invalidTagsAndAttributes.tags;
+              var attrs = invalidTagsAndAttributes.attrs;
+              if (tags.length === 0 && attrs.length === 0) {
+                var resampledFile = (
+                  ImageUploadHelperService.convertImageDataToImageFile(
+                    dataURI));
+                var date = new Date();
+                var now = date.getTime();
+                // This temporary Id will be used for adding and retrieving the
+                // raw image for each LaTeX string from the request body. For
+                // more details refer to the docstring in
+                // sendMathSvgsToBackend() in AdminBackendApiService.
+                var latexId = (
+                  now.toString(36).substr(2, 6) +
+                  Math.random().toString(36).substr(4));
+                resolve ({
+                  file: resampledFile,
+                  dimensions: {
+                    encoded_height_string: dimensions.height,
+                    encoded_width_string: dimensions.width,
+                    encoded_vertical_padding_string: dimensions.verticalPadding
+                  },
+                  latexId: latexId
+                });
+              } else {
+                AlertsService.addWarning('SVG failed validation.');
```

Comment:
Do you need to provide info about which one, for debugging purposes?

Program defensively so that you get all the info you need when we do a test run.

Reply:
Yeah, I've added more logging